sphinx/writers/latex.py

```diff
@@ -400,15 +400,26 @@ def get_colspec(self):
             total = sum(self.colwidths)
             colspecs = ['\\X{%d}{%d}' % (width, total) for width in self.colwidths]
             return '{|%s|}\n' % '|'.join(colspecs)
-        elif self.has_problematic:
-            return '{|*{%d}{\\X{1}{%d}|}}\n' % (self.colcount, self.colcount)
-        elif self.get_table_type() == 'tabulary':
-            # sphinx.sty sets T to be J by default.
-            return '{|' + ('T|' * self.colcount) + '}\n'
-        elif self.has_oldproblematic:
-            return '{|*{%d}{\\X{1}{%d}|}}\n' % (self.colcount, self.colcount)
+        elif not self.get_table_type() == 'tabulary':
```

Comment:
I feel positive condition is a bit simpler. How about `elif self.get_table_type() in ('longtable', 'tabular')` instead?